app/controllers/application_controller.rb

```diff
@@ -10,11 +10,19 @@ class ApplicationController < ActionController::API
 
   def with_current_request
     ManageIQ::API::Common::Request.with_request(request) do |current|
+      raise Catalog::NotAuthorized, "User not Entitled" unless check_entitled(current.entitlement)
```

Comment:
Maybe this should raise the `EntitlementError` that Drew defined in his PR [here](https://github.com/ManageIQ/manageiq-api-common/pull/59/files#diff-59c2b8ea17f87c8b355c52ee297dedceR4)

Reply:
Sure, that sounds good. updated